TEST

```diff
@@ -50,86 +53,94 @@ do_testing() {
     local n=0 success=0 fail=0 timeout=0 admissible=0 lexer_fail=0 replay_fail=0
 
     # Limit runtime of each test case, in seconds.
-    ulimit -t $TIMEOUT
+    ulimit -t "${TIMEOUT}"
 
     for i in $(test_cases | sort -n) ; do
         n=$((n+1))
-        echo -ne "$C[start]**$CEND TEST $n: $i ... "
-        local uhoh=0
+        echo -ne "${C[start]}**${CEND} TEST ${n}: ${i} ... "
 
-        result=$($LEX_CHECK "$i" 2>&1)
+        result=$(${LEX_CHECK} "${i}" 2>&1)
         exit_code=$?
 
-        if [ $exit_code -eq 2 ] ; then
+        if [ ${exit_code} -eq 2 ] ; then
             let lexer_fail=lexer_fail+1
-            echo -e "$C[lex_fail]FATAL LEXER FAILURE$CEND"
-            test $STOP -eq 1 && exit 1 || continue
+            echo -e "${C[lex_fail]}FATAL LEXER FAILURE${CEND}"
+            test ${STOP} -eq 1 && exit 1 || continue
         fi
 
-        if [ $exit_code -ne 0 ] ; then
+        if [ ${exit_code} -ne 0 ] ; then
             lexer_fail=$((lexer_fail+1))
-            echo -en "$C[lex_fail]LEXER FAILURE$CEND "
-            test $STOP -eq 1 && exit 1
+            echo -en "${C[lex_fail]}LEXER FAILURE${CEND}"
+            test ${STOP} -eq 1 && exit 1
         fi
 
-        result=$($EXE +test $* $i 2>&1)
+        result=$(${EXE} +test $* "${i}" 2>&1)
         exit_code=$?
-        diff_result=$(printf "%s" $result | diff -b -u $i.out - 2>/dev/null)
+        echo "abcd${result}efgh"
```

Comment:
Is this a debug print?

Reply:
@tsani Oh my. Yes, it's a debug print and I forgot it. I'm going to remove it right now.